ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -1783,48 +1783,54 @@ static void ggml_vk_load_shaders(vk_device& device) {
     }
 
     // mul mat vec
-    // computing two rows per workgroup is a benefit for Q4_0 -> Q5_1, but not for Q8_0.
+
+    // AMD GCN graphics cards perform best when the number of rows per shader is doubled
+    uint32_t rm = 1;
+    if ((device->vendor_id == VK_VENDOR_ID_AMD) && (device->subgroup_size == 64))
+        rm = 2;
```

Comment:
This doesn't affect just GCN, but also RDNA. The subgroup_size check is basically redundant cause all AMD GPUs (which support Vulkan) have a subgroup_size of 64.

I see a slight regression with this row change on my 6800 XT. I don't have time right now, but I'll investigate further tomorrow or so.

If we do have to distinguish GCN from RDNA one way might be #10721, cause GCN has min and max subgroup sizes of 64, while RDNA has min 32 and max 64.

Reply:
I've pulled in your https://github.com/ggerganov/llama.cpp/pull/10721 and used the min and max check to confirm that I'm using a GCN card. I think that's the only way to do it and it looks like there's nothing in Vulkan that tells you the precise architecture name of the graphics card.

In this case we'll need to merge this after https://github.com/ggerganov/llama.cpp/pull/10721.